### PR TITLE
No longer use 'throw()', use 'noexcept' instead.

### PR DESCRIPTION
**Pull request by bangerth**
This addresses the fact that 'throw' specifications are/will be deprecated.
In reference to #2882.

I verified that it compiles in both C++11 and C++03 mode.

### CODE REVIEW

File: source/base/exceptions.cc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // ---------------------------------------------------------------------
 //
-// Copyright (C) 1998 - 2014 by the deal.II authors
+// Copyright (C) 1998 - 2014, 2016 by the deal.II authors
```
Comment (Rombur) [inline source/base/exceptions.cc]: Copyright format is wrong.

**Comment by Rombur** [comment]
Looks good. Just fix the copyright date and merge yourself.

File: include/deal.II/base/config.h.in [hunk 2]
```diff
@@ -267,6 +267,23 @@
 
 
 /***********************************************************************
+ * A macro that is used to mark function as not throwing any exceptions.
+ * The appropriate way to do this in C++98 was to say
+ *     void foo () throw ();
```
Comment (drwells) [inline include/deal.II/base/config.h.in]: Is this right? I thought this sort of annotation said that a function *could* throw an exception, which would not be equivalent to `noexcept` (which cannot throw).

Comment (Rombur) [inline include/deal.II/base/config.h.in]: ```throw()``` cannot throw see http://stackoverflow.com/questions/12833241/difference-between-c03-throw-specifier-c11-noexcept and http://stackoverflow.com/questions/88573/should-i-use-an-exception-specifier-in-c

Comment (bangerth) [inline include/deal.II/base/config.h.in]: Correct. What's in the parentheses is a *list* of possible exception types. If the list is empty, then the function is not allowed to throw.

Comment (drwells) [inline include/deal.II/base/config.h.in]: Ah, got it.

**Pull request merged**